adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -500,15 +500,17 @@ - (void)vungleAdPlayabilityUpdate:(BOOL)isAdPlayable
   if (!delegate) {
     return;
   }
-  if (error) {
-    NSLog(@"Vungle Ad Playability returned an error: %@", error.localizedDescription);
-    [delegate adNotAvailable:error];
-    [self removeDelegate:delegate];
-    return;
-  }
 
   if (isAdPlayable) {
     [delegate adAvailable];
+  } else {
+    if (error) {
+      NSLog(@"Vungle Ad Playability returned an error: %@", error.localizedDescription);
+    } else {
+      error = GADMAdapterVungleErrorWithCodeAndDescription(kGADErrorMediationAdapterError, [NSString stringWithFormat:@"Ad is not available for placementID: %@.", placementID]);
```

Comment:
Might be a good idea to make sure you have a placement ID here, before including it in the error response.

Reply:
@crbinsf Got it, I will add a judgment statement here, thanks a lot for the guide. 😊

Reply:
@crbinsf Hi Clarke, I have checked the code, we don't need to check placementID here, if placementID if nil or invalid, the delegate returned `id<GADMAdapterVungleDelegate> delegate = [self getDelegateForPlacement:placementID];` will be `nil` in the top of this method, which makes this method return immediately. Thanks ^_^

Reply:
Ah - I missed that. Thank you for clarifying!